src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs

```diff
@@ -198,13 +200,48 @@ await requestContext.SendResult(new ResultStatus()
         /// Handles request for schema compare publish changes script
         /// </summary>
         /// <returns></returns>
-        public async Task HandleSchemaComparePublishChangesRequest(SchemaComparePublishChangesParams parameters, RequestContext<ResultStatus> requestContext)
+        public async Task HandleSchemaComparePublishChangesRequest(SchemaComparePublishDatabaseChangesParams parameters, RequestContext<ResultStatus> requestContext)
         {
-            SchemaComparePublishChangesOperation operation = null;
+            SchemaComparePublishDatabaseChangesOperation operation = null;
             try
             {
                 SchemaComparisonResult compareResult = schemaCompareResults.Value[parameters.OperationId];
-                operation = new SchemaComparePublishChangesOperation(parameters, compareResult);
+                operation = new SchemaComparePublishDatabaseChangesOperation(parameters, compareResult);
+                SqlTask sqlTask = null;
+                TaskMetadata metadata = new TaskMetadata();
+                metadata.TaskOperation = operation;
+                metadata.ServerName = parameters.TargetServerName;
+                metadata.DatabaseName = parameters.TargetDatabaseName;
+                metadata.Name = SR.PublishChangesTaskName;
+                sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
```

Comment:
should be able to declare sqlTask here instead of initially being set to null above